src/js/components/DateInput/DateInput.js

```diff
@@ -172,42 +177,51 @@ const DateInput = forwardRef(
           onEsc={open ? () => closeCalendar() : undefined}
           onSpace={openCalendar}
         >
-          <MaskedInput
-            ref={ref}
-            id={id}
-            name={name}
-            icon={<CalendarIcon size={iconSize} />}
-            reverse
-            disabled={disabled}
-            mask={mask}
-            {...inputProps}
-            {...rest}
-            value={textValue}
-            onChange={(event) => {
-              const nextTextValue = event.target.value;
-              setTextValue(nextTextValue);
-              const nextValue = textToValue(
-                nextTextValue,
-                schema,
-                value,
-                range,
-              );
-              // update value even when undefined
-              setValue(nextValue);
-              if (onChange) {
-                event.persist(); // extract from React synthetic event pool
-                const adjustedEvent = event;
-                adjustedEvent.value = nextValue;
-                onChange(adjustedEvent);
-              }
-            }}
-            onFocus={(event) => {
-              announce(
-                formatMessage({ id: 'dateInput.openCalendar', messages }),
-              );
-              if (onFocus) onFocus(event);
-            }}
-          />
+          <Box border={!plain} round="xxsmall" direction="row" fill>
+            <MaskedInput
+              ref={ref}
+              id={id}
+              name={name}
+              reverse
+              disabled={disabled}
+              mask={mask}
+              plain
+              {...inputProps}
+              {...rest}
+              value={textValue}
+              onChange={(event) => {
+                const nextTextValue = event.target.value;
+                setTextValue(nextTextValue);
+                const nextValue = textToValue(
+                  nextTextValue,
+                  schema,
+                  value,
+                  range,
+                );
+                // update value even when undefined
+                setValue(nextValue);
+                if (onChange) {
+                  event.persist(); // extract from React synthetic event pool
+                  const adjustedEvent = event;
+                  adjustedEvent.value = nextValue;
+                  onChange(adjustedEvent);
+                }
+              }}
+              onFocus={(event) => {
+                announce(
+                  formatMessage({ id: 'dateInput.openCalendar', messages }),
+                );
+                if (onFocus) onFocus(event);
+              }}
+            />
+            <Button
+              onClick={open ? closeCalendar : openCalendar}
```

Comment:
Every time I click, it is opening the calendar instead of opening and closing -- maybe removing the callback wasn't the right call.

Can we do:
```
haven't tried this locally
() => open ? closeCalendar() : openCalendar()

OR the original approach

open ? () => closeCalendar() : () => openCalendar()
```

Reply:
Or is having it always open on click the desired behavior? That seems to be the behavior in the native HTML: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date

if that's the case, we could just to onClick={() =>openCalendar}

Reply:
This was something I was debating. Even though the native HTML date input always opens on click I was leaning towards having the button open and close the drop.

Reply:
When I try `() => open ? closeCalendar() : openCalendar()` I get the following error: `Too many re-renders. React limits the number of renders to prevent an infinite loop.`

Going to go with `open ? () => closeCalendar() : () => openCalendar()`, this appears to be working correctly

Reply:
What about: `open ? closeCalendar : openCalendar`?

Reply:
`open ? closeCalendar : openCalendar` was resulting in the calendar only opening instead of opening and closing

Reply:
After changing the onClickOutside function `open ? closeCalendar : openCalendar` works fine. Switching to use this and pushing up changes now.